src/prefect/results.py

```diff
@@ -226,6 +238,67 @@ async def update_for_task(self: Self, task: "Task") -> Self:
             update["storage_block"] = await get_default_result_storage()
         return self.model_copy(update=update)
 
+    @sync_compatible
+    async def read(self, key: str) -> "ResultRecord":
```

Comment:
two cents: I think we should start adhering to the sync/async naming convention on new methods; to avoid making this a whole rewrite, i'd do:
```python
@sync_compatible
async def _load(...):
    # all the logic you have here already
    
def load(...):
    return self._load(_sync=True)

async def aload(...):
    return self._load(_sync=False)
```

or something to that effect.

I won't hold up the PR for it though if that introduces edge cases

Reply:
Makes sense! I'll make that change